### PR TITLE
Sync notebooks

### PR DESCRIPTION
**Pull request by andypetrella**
Two notebooks opened on different browsers will process web socket events for
- [x] results
- [x] observable
- [ ] new cell
- [ ] code change

The last two are trickier because there are no events for them...

> This requires web socket connection opened to the server

/cc @vidma @meh-ninja @virtualirfan

### CODE REVIEW

**Comment by andypetrella** [comment]
The last two points aren't for this PR, they're just mentioned for sake of completeness.

So to me it's ready to :ship: .

# Check out 
Using two tabs/browser opened on the same notebook, execute some cells

## Simple Cell
```scala
val a = "ok"
```

## Getting fancy with plots
It's way fancier with observables

```scala
val l = LineChart((1 to 10).toList)
```
And execute twice the following cell (one time on each tab)
```scala
l.addAndApply((11 to 20).toList)
```

## Spark too
Spark notification will work fine too, create cells for the following and execute them:
```scala
val rdd = sparkContext.parallelize(1 to 1000)
```
and
```scala
val r = rdd.repartition(100).flatMap(i => 1 to 1000).repartition(1000).groupBy(_ % 2).countByValue
r.head._1._2.toList
```

**Comment by meh-ninja** [comment]
:+1:

**Comment by vidma** [comment]
good start (didn't look very carefully).

File: modules/observable/src/main/scala/notebook/ObsWebSocketService.scala [hunk 1]
```diff
@@ -19,19 +19,35 @@ class ObsWebSocketService(system: ActorSystem, val channel: Concurrent.Channel[J
...
+      case m@("add", channel: Concurrent.Channel[JsValue]) =>
+        //println(s"Adding channel $channel to channels $channels")
+        channels = channel :: channels
+
+      case m@("remove", channel: Concurrent.Channel[JsValue]) =>
+        //println(s"Removing channel $channel to channels $channels")
+        channels = channels.filter(_ != channel)
+        if (channels.isEmpty) {
+          //println(s"Killing WebSocketObservableActor since no more channels are attached")
+          sender() ! akka.actor.PoisonPill
```
Comment (vidma) [inline modules/observable/src/main/scala/notebook/ObsWebSocketService.scala]: what does this kill? and is it intentional? 
what if it's a page refresh? what if user will reopen the notebook soon?
can it be related to #531 ?

Comment (vidma) [inline modules/observable/src/main/scala/notebook/ObsWebSocketService.scala]: yes it is. fixed in #533

**Pull request merged**